Apps/FlightSoftware/Watchdog2/src/i2c.c

```diff
@@ -0,0 +1,318 @@
+#include "include/i2c.h"
+
+// TODO: integrate this code with the rest of the system
+
+int8_t raw_battery_charge[2];
+int8_t raw_battery_voltage[2];
+int8_t raw_battery_current[2];
+int8_t raw_fuel_gauge_temp[2];
+
+uint8_t batt_charge_telem;
+uint8_t batt_curr_telem;
+
+uint8_t fuel_gauge_write_control_reg;
+uint8_t fuel_gauge_read_control_reg;
+
+/*
+ * File for interfacing with I2C protocol hardware module
+ *
+ * I2C communication uses eUSCI_B0 (power circuitry)
+ */
+
+/* Used to track the state of the software state machine*/
+I2C_Mode MasterMode = IDLE_MODE;
+
+/* The Register Address/Command to use*/
+uint8_t TransmitRegAddr = 0;
+
+// initialize buffers and their corresponding byte counters
+uint8_t ReceiveBuffer[I2C_RX_BUFFER_MAX_SIZE] = {0};
+uint8_t RXByteCtr = 0;
+uint8_t ReceiveIndex = 0;
+uint8_t TransmitBuffer[I2C_TX_BUFFER_MAX_SIZE] = {0};
+uint8_t TXByteCtr = 0;
+uint8_t TransmitIndex = 0;
```

Comment:
I think the interrupts and buffers used here should be reworked, but at a bare minimum (if nothing else is changed) all of these variables need to be volatile as they are modified and/or read both in and out of the ISR